data/spells/scripts/summon/summons.lua

```diff
@@ -36,7 +36,7 @@ function onCastSpell(cid, var)
 
     if not summonName then return false end
 
-    local mySummon = Game.createMonster(summonName, player:getPosition(), true, false)
+    local mySummon = Game.createMonster(summonName, player:getPosition(), true, false, creature)
```

Comment:
```suggestion
    local mySummon = Game.createMonster(summonName, player:getPosition(), true, false, cid)
```
either we have to change it to "cid" or whole function to `onCastSpell(creature, variant)`

Reply:
i'll change the whole function to creature/variant
Thanks